### DISCUSSION TITLE
iOS 18 REGRESSION: Pinch and drag gestures close the AR QuickLook session window

### DISCUSSION THREAD

**Discussion by milesgreen** [google/model-viewer#4841] [General]
Hi @AdaRoseCannon - another iOS regression for you ❤️

**Observations:**

When viewing a 3D model in AR QuickLook, some gesture interactions cause the entire ARQL window to start minimising and eventually closing the session window altogether. 

If user performs a pinch-close gesture to scale the model down, the entire ARQL window starts to shrink and eventually closes. 

But user _is_ able to perform a pinch-open gesture to scale model up, and this works fine. 

Also, if user performs a drag down gesture, to move the model closer on the horizontal plane, the ARQL window shrinks and closes. 

But user _is_ able to perform drag up / left / right gestures to move model, and this works fine. 

**Expected Results:**

User should be able to pinch-close (to scale down) or drag downwards (to reposition model) without exiting ARQL session. 

**Repro steps:**

* Using iOS 18 beta
* In Safari, visit: https://developer.apple.com/augmented-reality/quick-look/
* Select a model to view in ARQL
* Once model is anchored, pinch model to scale down
* Observe the model does not shrink, but the entire ARQL window shrinks
* Continuing to pinch will eventually minimise the ARQL session window altogether, returning user to originating webpage
* Alternatively, open 3D model, once anchored, drag downwards to reposition, and observe ARQL screen minimises and closes

Note that when performing the same interactions with third party browsers such as Chrome and Edge, the issue is not observed.

Issue reported via Feedback Assistant (FB14470881)

Issue reported via Webkit Bugzilla (https://bugs.webkit.org/show_bug.cgi?id=277012)

https://github.com/user-attachments/assets/73f229f0-d1e4-4e05-b992-e3ea99727512

**Comment by milesgreen** [comment]
Retesting with iOS 18.0 Dev Beta 5 (22A5326f)

Behaviour has changed with latest beta - but is not completely resolved. ARQL window no longer shrinks / minimises. But the gestures still fail to work as expected.  

Current observations:

- When model opens and anchors in ARQL, pinch close gesture now does nothing - the 3D model does not scale down.

- Pinch open gesture does work - the model scales up. 

- If you first pinch open and, within the same gesture, then pinch close, the model can be scaled down. 

- Similarly, drag down gesture does nothing - the 3D model does not move closer. 

- Drag up / left / right gestures work - the model is moved on the anchored plane. 

- If you first drag up / left / right and then, within the same gesture, drag down, the model can be moved closer.

**Comment by milesgreen** [comment]
Retesting with iOS 18.0 Dev Beta 8 (22A5350a)
Issue is still not completely resolved. 
Pinch close gesture does not scale down 3D model.
Drag down gesture does not move 3D model.

**Comment by AdaRoseCannon** [comment]
Please can you file a new Feedback Assistant to say it's not completely resolved.

**Reply by milesgreen** [reply]
OK @AdaRoseCannon , new case filed on Feedback Assistant (FB15025959)
Thanks!

**Comment by milesgreen** [comment]
This didn't get resolved in iOS 18.0 public release. 

But the good news is that I'm retesting it in iOS 18.1 beta (22B5069a), and it looks to be fixed!

Pinch close gesture now does scale down 3D model.
Drag down gesture now does move 3D model.
